sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorSession.java

```diff
@@ -88,6 +96,38 @@
         return createConsumer(linkName, entityPath, entityType, timeout, retry, receiveMode, filter);
     }
 
+    @Override
+    public Mono<AmqpLink> createProducer(String linkName, String entityPath, Duration timeout,
+        AmqpRetryPolicy retry, String transferDestinationPath) {
+        Objects.requireNonNull(entityPath, "'entityPath' cannot be null.");
+        Objects.requireNonNull(timeout, "'timeout' cannot be null.");
+        Objects.requireNonNull(retry, "'retry' cannot be null.");
+
+        final Duration serverTimeout = adjustServerTimeout(timeout);
+        Map<Symbol, Object> linkProperties = new HashMap<>();
+
+        linkProperties.put(LINK_TIMEOUT_PROPERTY, UnsignedInteger.valueOf(serverTimeout.toMillis()));
+
+        if (!CoreUtils.isNullOrEmpty(transferDestinationPath)) {
+            linkProperties.put(LINK_TRANSFER_DESTINATION_PROPERTY, transferDestinationPath);
+            logger.verbose("Get or create sender link {} for via entity path: '{}'", linkName, entityPath);
+
+            final TokenManager tokenManager = tokenManagerProvider.getTokenManager(cbsNodeSupplier,
+                transferDestinationPath);
+
+            return tokenManager.authorize()
+                .then(
```

Comment:
Why create an entire Mono for this? You can use .doFinally. authorize() is a Mono.